vendor_data.py

```diff
@@ -1,241 +1,375 @@
 import argparse
-import re
-import pandas as pd
-import os
-import pyinputplus as pyip
-
-def change_dictionary_value(dictionary, key, index, new_value):
-    if key in dictionary and index in dictionary[key]:
-        dictionary[key][index] = new_value
-        return True
-    else:
-        return False
+import re #regular expression package to validate e-mail
+import pandas as pd #pandas package to write excel file
+import os #os package to check if files exists
+import pyinputplus as pyip #pyinputplus package to facilitate providing arguments
+
+#4 methods to make validation easier
+def ChangeDictionaryValue(Dictionary, Key, Index, NewValue):
+    """ChangeDictionaryValue, change values of a dictionary
+
+    Args:
+        Dictionary (dict): name of the dictionary that will be read
+        Key (str): the key for which the value will be changed
+        Index (int): the index of the value that will be changed
+        NewValue: new value for the chosen index, type depends on the key
+    """
+    if Key in Dictionary and Index in Dictionary[Key]:
+        Dictionary[Key][Index] = NewValue
+def CheckVendorMail(VendorMail):
+    """CheckVendorMail, validate e-mail address
+
+    Args:
+        VendorMail (str): an e-mail address
+
+    Raises:
+        TypeError: if the argument provided is not a string, a TypeError is raised
+        ValueError: if the address provided does not conform with the regex string `regex_mail`, a ValueError is raised
+    """
+    regex_mail = "^[a-z0-9]+[\._]?[a-z0-9]+[@]\w+[.]\w{2,3}$" #regex used to validate email
+    if type(VendorMail) != str:
+        raise TypeError ("VendorMail should be a string!")
+    if not(re.search(regex_mail,VendorMail)):
+        raise ValueError("Please insert a valid email address.")
+def CheckWordRate(WordRate):
+    """CheckWordRate, validate word rate
 
+    Args:
+        WordRate (float): a word rate in EUR/word
+
+    Raises:
+        TypeError:  if the argument provided is not a float, a TypeError is raised
+        ValueError: the argument provided should not be higher than 0.15
+        ValueError: the argument provided should not be 0.00
+    """
+    if type(WordRate) != float:
+        raise TypeError("WordRate should be a float!")
+    if WordRate > 0.15:
+        raise ValueError("This vendor is too expensive, pick another one.")
+    if WordRate == 0.00:
+        raise ValueError("Word rate cannot be 0.00.")
+def CheckCatTool(CatTool):
+    """CheckCatTool, validate CatTOol
+
+    Args:
+        CatTool (str): a CAT Tool
+
+    Raises:
+        TypeError: if the argument provided is not string, a TypeError is raised
+        ValueError: the argument provided should be one of the four CAT Tools in the list ["XTM", "Trados Studio", "MemoQ", "Memsource"]
+    """
+    if type(CatTool) != str:
+        raise TypeError("CatTool should be a string!")
+    if not CatTool in ["XTM", "Trados Studio", "MemoQ", "Memsource"]:

```

Comment:
To avoid repetition (if you wanted to add/remove a CatTool, now you have to do it in two places) either provide the list of CatTools as an argument of this function or call it as `VendorData.CatTools`.